internal/querycoord/cluster.go

```diff
@@ -376,6 +378,21 @@ func (c *queryNodeCluster) getSegmentInfo(ctx context.Context, in *querypb.GetSe
 	return segmentInfos, nil
 }
 
+func (c *queryNodeCluster) getSegmentInfoByNode(ctx context.Context, nodeID int64, in *querypb.GetSegmentInfoRequest) ([]*querypb.SegmentInfo, error) {
+	c.RLock()
+	defer c.RUnlock()
+
+	if node, ok := c.nodes[nodeID]; ok {
+		res, err := node.getSegmentInfo(ctx, in)
+		if err != nil {
+			return nil, err
+		}
+		return res.Infos, nil
+	}
+
+	return nil, errors.New("ReleasePartitions: can't find query node by nodeID")
```

Comment:
Error message is weird, NodeID should. be printed

Reply:
fixed